packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.stories.js

```diff
@@ -44,3 +45,13 @@ export const ScrollDown = () => {
     </Box>
   )
 }
+
+export const DeferredAnimation = () => {
+  const [value, setValue] = useState(0)
+  setTimeout(() => setValue(700000000), 2000)
+  return (
+    <Box pad={{ vertical: '120vh' }}>
+      <AnimatedNumber duration={4000} value={value} />
+    </Box>
+  )
+}
```

Comment:
If you checkout the first commit on this branch, you can use this story to reproduce the original bug. On the main branch, this story always displays 0.